### PR TITLE
utils_version: Update to logic to check version in the interval

### PR DESCRIPTION
**Pull request by pingl2016**
In python3, comparing non-numerical objects of different type raises
a TypeError by default. Correct the logic to check the version in
the interval when the lower boundary or upper boundary doesn't
exist.

Signed-off-by: Ping Li <pingl@redhat.com>

ID: 1608691

### CODE REVIEW

**Comment by pingl2016** [comment]
# python3 
Python 3.6.5 (default, May 10 2018, 08:50:37) 
[GCC 8.1.1 20180502 (Red Hat 8.1.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from virttest.utils_version import VersionInterval
>>> "2.10.7" in VersionInterval("(2.1.2,)")
True
>>> "1.5.3" in VersionInterval("(2.1.2,)")
False
>>> "2.1.2" in VersionInterval("[2.1.2,)")
True
>>> 
>>> "2.10.7" in VersionInterval("(, 2.3.2)")
False
>>> "2.3.1" in VersionInterval("(, 2.3.2)")
True
>>> "2.3.2" in VersionInterval("(, 2.3.2]")
True
>>> 
>>> "2.1.2" in VersionInterval("(2.1.2, 2.10.1)")
False
>>> "2.8.0" in VersionInterval("(2.1.2, 2.10.1)")
True
>>> "2.10.7" in VersionInterval("(2.1.2, 2.10.1)")
False
>>> "2.10.1" in VersionInterval("[2.1.2, 2.10.1]")
True
>>> "2.1.2" in VersionInterval("[2.1.2, 2.10.1]")

**Comment by pingl2016** [comment]
# python
Python 2.7.5 (default, Jun 12 2018, 10:42:23) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-34)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from virttest.utils_version import VersionInterval
>>> "2.10.7" in VersionInterval("(2.1.2,)")
True
>>> "1.5.3" in VersionInterval("(2.1.2,)")
False
>>> "2.1.2" in VersionInterval("[2.1.2,)")
True
>>> 
>>> "2.10.7" in VersionInterval("(, 2.3.2)")
False
>>> "2.3.1" in VersionInterval("(, 2.3.2)")
True
>>> "2.3.2" in VersionInterval("(, 2.3.2]")
True
>>> 
>>> "2.1.2" in VersionInterval("(2.1.2, 2.10.1)")
False
>>> "2.8.0" in VersionInterval("(2.1.2, 2.10.1)")
True
>>> "2.10.7" in VersionInterval("(2.1.2, 2.10.1)")
False
>>> "2.10.1" in VersionInterval("[2.1.2, 2.10.1]")
True
>>> "2.1.2" in VersionInterval("[2.1.2, 2.10.1]")
True

**Comment by pingl2016** [comment]
(1/3) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.cluster_size_check.positive_testing: PASS (2.91 s)
 (2/3) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.cluster_size_check.negative_testing.cluster_size_0k: PASS (2.63 s)
 (3/3) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.cluster_size_check.negative_testing.cluster_size_4M: PASS (2.62 s)

File: virttest/utils_version.py [hunk 1]
```diff
@@ -28,8 +28,10 @@ def __init__(self, interval):
             raise ValueError("Invaild string representation of an interval")
         self.opening, lower, upper, self.closing = match.groups()
 
-        self.lower_bound = LooseVersion(lower) if lower else LooseVersion("0")
-        self.upper_bound = LooseVersion(upper) if upper else LooseVersion("z")
+        self.lower_bound = LooseVersion(lower) if lower else None
+        self.upper_bound = LooseVersion(upper) if upper else None
+        if self.upper_bound or self.lower_bound:
+            return
```
Comment (luckyh) [inline virttest/utils_version.py]: It does need a `ValueError` check when both have `self.upper_bound` and `self.lower_bound`.

Comment (pingl2016) [reply virttest/utils_version.py]: Do you refer to both lower boundary and upper boundary are None?

**Comment by pingl2016** [comment]
\# python3
Python 3.6.5 (default, May 10 2018, 08:50:37) 
[GCC 8.1.1 20180502 (Red Hat 8.1.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
\>>> from virttest.utils_version import VersionInterval
\>>> "2.10.7" in VersionInterval("(2.1.2,)")
True
\>>> "2.1.2" in VersionInterval("(2.1.2,)")
False
\>>> "2.1.2" in VersionInterval("[2.1.2,)")
True
\>>> 
\>>> "2.3.2" in VersionInterval("(, 2.3.2)")
False
\>>> "1.5.3" in VersionInterval("(, 2.3.2)")
True
\>>> "2.3.2" in VersionInterval("(, 2.3.2]")
True
\>>> 
\>>> "2.1.2" in VersionInterval("(2.1.2, 2.10.1)")
False
\>>> "2.10.1" in VersionInterval("(2.1.2, 2.10.1)")
False
\>>> "2.8.0" in VersionInterval("(2.1.2, 2.10.1)")
True
\>>> "2.10.1" in VersionInterval("[2.1.2, 2.10.1]")
True
\>>> "2.1.2" in VersionInterval("[2.1.2, 2.10.1]")
True
\>>> "2.1.2" in VersionInterval("[2.1.2, 2.1.2]")
True
\>>> "2.1.2" in VersionInterval("(2.1.2, 2.1.2]")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/kar/workspace/avocado-vt/virttest/utils_version.py", line 38, in __init__
    raise ValueError("Invaild interval")
ValueError: Invaild interval
\>>> "2.1.2" in VersionInterval("[2.1.2, 2.1.2)")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/kar/workspace/avocado-vt/virttest/utils_version.py", line 38, in __init__
    raise ValueError("Invaild interval")
ValueError: Invaild interval

**Comment by pingl2016** [comment]
\# python
Python 2.7.5 (default, Jun 12 2018, 10:42:23) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-34)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
\>>> from virttest.utils_version import VersionInterval
\>>> "2.10.7" in VersionInterval("(2.1.2,)")
True
\>>> "2.1.2" in VersionInterval("(2.1.2,)")
False
\>>> "2.1.2" in VersionInterval("[2.1.2,)")
True
\>>> 
\>>> "2.3.2" in VersionInterval("(, 2.3.2)")
False
\>>> "1.5.3" in VersionInterval("(, 2.3.2)")
True
\>>> "2.3.2" in VersionInterval("(, 2.3.2]")
True
\>>> 
\>>> "2.1.2" in VersionInterval("(2.1.2, 2.10.1)")
False
\>>> "2.10.1" in VersionInterval("(2.1.2, 2.10.1)")
False
\>>> "2.8.0" in VersionInterval("(2.1.2, 2.10.1)")
True
\>>> "2.10.1" in VersionInterval("[2.1.2, 2.10.1]")
True
\>>> "2.1.2" in VersionInterval("[2.1.2, 2.10.1]")
True
\>>> "2.1.2" in VersionInterval("[2.1.2, 2.1.2]")
True
\>>> "2.1.2" in VersionInterval("(2.1.2, 2.1.2]")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "virttest/utils_version.py", line 38, in __init__
    raise ValueError("Invaild interval")
ValueError: Invaild interval
\>>> "2.1.2" in VersionInterval("[2.1.2, 2.1.2)")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "virttest/utils_version.py", line 38, in __init__
    raise ValueError("Invaild interval")
ValueError: Invaild interval

**Comment by pingl2016** [comment]
(1/3) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.cluster_size_check.positive_testing: PASS (2.84 s)
 (2/3) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.cluster_size_check.negative_testing.cluster_size_0k: PASS (2.45 s)
 (3/3) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.cluster_size_check.negative_testing.cluster_size_4M: PASS (2.51 s)

**Comment by pingl2016** [comment]
@luckyh @hereischen updated, please help to check it again

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

File: virttest/utils_version.py [hunk 2]
```diff
@@ -28,10 +29,19 @@ def __init__(self, interval):
...
 
-        self.lower_bound = LooseVersion(lower) if lower else LooseVersion("0")
-        self.upper_bound = LooseVersion(upper) if upper else LooseVersion("z")
-        if self.lower_bound > self.upper_bound:
-            raise ValueError("Invaild interval")
+        self.lower_bound = LooseVersion(lower) if lower else None
+        self.upper_bound = LooseVersion(upper) if upper else None
+        self._check_interval()
+
+    def _check_interval(self):
```
Comment (hereischen) [inline virttest/utils_version.py]: maybe `_validate_interval()` or `_validator `, but the current is ok to me

**Pull request merged**